sb3_contrib/tqc/tqc.py

```diff
@@ -242,19 +223,18 @@ def train(self, gradient_steps: int, batch_size: int = 64) -> None:
                 next_actions, next_log_prob = self.actor.action_log_prob(replay_data.next_observations)
                 # Compute and cut quantiles at the next state
                 # batch x nets x quantiles
-                next_z = self.critic_target(replay_data.next_observations, next_actions)
-                sorted_z, _ = th.sort(next_z.reshape(batch_size, -1))
-                sorted_z_part = sorted_z[:, : self.critic.quantiles_total - top_quantiles_to_drop]
+                next_quantile = self.critic_target(replay_data.next_observations, next_actions)
```

Comment:
dito (about singular/plural)

Reply:
>probably rename it quantile_backup so you don't overwrite target_quantile defined above?

(Belleman) "buckup" and "target" mean the same thing here, don't they?
For clarity, it may be good to name them `next_*` and `target_*`? (it would be incompatible with stable_baselines3)
What do you think?
```
next_* = ...  # Estimate values for next observation
target_* = ... next_* ...   # Something like "R + gamma * next_*", which is the target for the loss
```

>btw, why target_quantile (singular) and not target_quantiles ?

I was confused that some variables like `next_log_prob`, `current_q`, and `target_q` are singular in DQN and SAC, but others are plural. There was no particular intention, so I'd be happy to rename variables as you prefer :)

Reply:
>(Belleman) "buckup" and "target" mean the same thing here, don't they?

yes, I was using "target" in SAC/TD3 because the next q-value came from the target network.
but I would agree with renaming `target_` to `next_` (and maybe rename the first `next_target_quantiles`), what do you think?

(we would need to do that in https://github.com/DLR-RM/stable-baselines3/pull/245 to keep SB3 in sync)

>current_q, and target_q are singular in DQN and SAC, 

sorry for that, I guess the true name should be `current_q_estimates` (plural), maybe a thing to change in https://github.com/DLR-RM/stable-baselines3/pull/245 (by @erniejunior )
and `log_prob` is short for "log probabilities" (I truncated it to avoid too long names and equations on several lines) but I agree it is a bit confusing.

Reply:
> I was using "target" in SAC/TD3 because the next q-value came from the target network.
> but I would agree with renaming target_ to next_ (and maybe rename the first next_target_quantiles), what do you think?

I see, now it makes sense. But I assume people call it "target" because it's the target value for training, not because it comes from the target network.
In my opinion, because "buckup" is a less common word, it's clearer for me either
1. to name both "Q(s_t+1)" and "R + gamma * (max)Q(s_t+1)"  as `target_*`, as in [DQN's code](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/dqn/dqn.py#L157)
2. to name "Q(s_t+1)" as `next_*` and "R + gamma * (max)Q(s_t+1)" as `target_*`, as in (my) [TQC's code](https://github.com/ku2482/stable-baselines3-contrib/blob/feat/qrdqn/sb3_contrib/tqc/tqc.py#L220).

> sorry for that, I guess the true name should be current_q_estimates (plural)

I see. So it's natural to name `current_quantile` or `current_quantile_estimates`?

Reply:
I would go for 2. (and we will need to change that in https://github.com/DLR-RM/stable-baselines3/pull/245.)

>I see. So it's natural to name current_quantile or current_quantile_estimates?

here it is slightly different as you have several quantiles (the complete name of `current_q_estimates` would be `current_q_values_estimates` maybe `current_q_values` would be better in fact)

Reply:
>here it is slightly different as you have several quantiles (the complete name of current_q_estimates would be current_q_values_estimates maybe current_q_values would be better in fact)

I see. I'm not fluent in English, so I'll follow your preference.

Reply:
I pushed some commits to rename them, what do you think?
(I modified TQC's code somewhat to make it easy to understand, sorry about that...)

Also, how should I evaluate QR-DQN?
The evaluation protocol of sb3 seems different from the paper.
(The paper reported the best agent's score within 125k steps(500k frames))

Anyway, I really appreciate your kind support ;)